contracts/src/periphery/cfg-getter/EigenDAConfigRetriever.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+import {IEigenDAConfigRegistry} from "src/core/interfaces/IEigenDADirectory.sol";
+import {ConfigRegistryTypes} from "src/core/libraries/v3/config-registry/ConfigRegistryTypes.sol";
+import {ConfigRegistryLib} from "src/core/libraries/v3/config-registry/ConfigRegistryLib.sol";
+
+/// @title EigenDAConfigRetriever
+/// @notice A stateless contract that defines getter functions using the underlying configuration registry's primitive API.
+contract EigenDAConfigRetriever {
+    IEigenDAConfigRegistry public immutable configRegistry;
+
+    constructor(address configRegistryAddress) {
+        configRegistry = IEigenDAConfigRegistry(configRegistryAddress);
+    }
+
+    /// @notice Retrieves all bytes32 configuration checkpoints for a given name that have an activation key greater than the provided activation key.
+    function getAllBytes32ConfigsAfterKey(string memory name, uint256 activationKey)
```

Comment:
This is not what we actually need.

We need to get the currently active value, plus all future values. This is only getting all future values.

In english, what we're trying to accomplish is:

"get the currently active configuration value, and also tell me all upcoming updates so I can plan ahead and make the config switch at the right moment"

Reply:
> I think the code is actually doing the right thing though?

Yeah, appears so 🤷 But also the function names are misleading

In any case, this code should probably be considered a draft for now. Once it's been reconsidered from first principles, and the docs + function names have been updated to be actually correct, then we can move forward with it.

For now, I'm converting it to `draft` mode